hightouch_blueprints/verify_sync.py

```diff
@@ -0,0 +1,128 @@
+import argparse
+import sys
+import requests
+import shipyard_utils as shipyard
+import errors
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--access-token', dest='access_token', required=True)
+    parser.add_argument('--sync-id', dest='sync_id', required=True)
+    parser.add_argument('--sync-run-id', dest='sync_run_id', required=False)
+    args = parser.parse_args()
+    return args
+
+
+
+def get_sync_status(sync_id, sync_run_id, access_token):
+    """
+    Gets the sync status from the Hightouch API
+    see: https://hightouch.io/docs/api-reference/#operation/ListSyncRuns
+    """
+    sync_api = f"https://api.hightouch.io/api/v1/syncs/{sync_id}/runs"
+    api_headers = {
+        'authorization': f"Bearer {access_token}",
+        'Content-Type': 'application/json'
+    }
+    params = {
+        "runId": sync_run_id
+    }
+    try:
+        sync_run_response = requests.get(sync_api,
+                                              params=params,
+                                              headers=api_headers)
+        
+        sync_status_code = sync_run_response.status_code
+        sync_run_json = sync_run_response.json()
+        # check if successful, if not return error message
+        if sync_status_code == requests.codes.ok:
+            return sync_run_json
+            
+        elif sync_status_code == 400: # Bad request
+            print("Sync status request failed due to Bad Request Error.")
+            sys.exit(errors.EXIT_CODE_BAD_REQUEST)
+            
+        elif sync_status_code == 401: # Incorrect credentials 
+            print("Incorrect credentials, check your access token")
+            sys.exit(errors.EXIT_CODE_INVALID_CREDENTIALS)
+            
+        elif sync_status_code == 422: # Invalid status query
+            print(f"Check status Validation failed: {sync_run_json['details']}")
+            sys.exit(errors.EXIT_CODE_BAD_REQUEST)
+        
+        else:
+            print("Unknown Error when sending request")
+            sys.exit(errors.EXIT_CODE_UNKNOWN_ERROR)
+        
+    except Exception as e:
+        print(f"Sync status for Sync {sync_id}:{sync_run_id} failed due to: {e}")
+        sys.exit(errors.EXIT_CODE_UNKNOWN_ERROR)
```

Comment:
Same issue as the other PR. Status codes aren't being properly assessed and it jumps straight to the exception with a message of `Expecting value: line 1 column 1 (char 0)`

Reply:
It took a bit to realize, but i'd placed the response.json() outside the success function. Not all error responses return a json response, some return plain text or just the status code and nothing else. 

I've gone through both rudderstack and hightouch to see if there were any other instances were i did this and fixed them